src/openforms/api/drf_spectacular/hooks.py

```diff
@@ -65,3 +67,36 @@ def add_middleware_headers(result, generator, request, public):
     )
 
     return result
+
+
+CSRF_TOKEN_PARAMETER = build_parameter_type(
+    name=CSRF_TOKEN_HEADER_NAME,
+    schema={"type": "string"},
+    location=OpenApiParameter.HEADER,
+    required=True,
+)
+
+
+def add_unsafe_methods_parameter(result, generator, request, public):
+    """
+    Schema generator hook to add parameters to endpoint with unsafe methods.
+    """
+    for path in result["paths"].values():
+        for operation_method, operation in path.items():
+            if operation_method not in ["post", "put", "patch", "delete"]:
+                continue
+
+            if "security" not in operation:
+                continue
+
+            for item in operation["security"]:
+                if "cookieAuth" in item:
+                    break
```

Comment:
I think the detection can be a bit more generic and not necessarily rely on the name, but rather if `type == apiKey` and `in == cookie`